# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContext2D.h
# Commit: 7b82837e2f56
# Full Hash: 7b82837e2f56582e1e86897285f4109856b72000
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2021-11-11 15:25:44
# Regressor Bug: 1739448
# File Overlap Count: 1
# Description:
#   Bug 1739448 - Implement a prototype WebGL-backed Canvas2D. r=gfx-reviewers,nical,jrmuizel
#   
#   This mainly provides DrawTargetWebgl, which implements the subset of the DrawTarget
#   API necessary for integration with CanvasRenderingContext2D. It translates them to
#   suitable commands for its internal ClientWebGLContext, which then manages remoting
# ==============================================================================

diff -r feac99f1dae3 -r 7b82837e2f56 dom/canvas/CanvasRenderingContext2D.h
--- a/dom/canvas/CanvasRenderingContext2D.h	Thu Nov 11 04:33:50 2021 +0000
+++ b/dom/canvas/CanvasRenderingContext2D.h	Thu Nov 11 04:33:50 2021 +0000
@@ -611,12 +611,18 @@
 
   void RestoreClipsAndTransformToTarget();
 
+  bool TryAcceleratedTarget(
+      RefPtr<gfx::DrawTarget>& aOutDT,
+      RefPtr<layers::PersistentBufferProvider>& aOutProvider);
+
   bool TrySharedTarget(RefPtr<gfx::DrawTarget>& aOutDT,
                        RefPtr<layers::PersistentBufferProvider>& aOutProvider);
 
   bool TryBasicTarget(RefPtr<gfx::DrawTarget>& aOutDT,
                       RefPtr<layers::PersistentBufferProvider>& aOutProvider);
 
+  ClientWebGLContext* AsWebgl() override;
+
   void RegisterAllocation();
 
   void SetInitialState();